v3/onesignal-notification.php

```diff
@@ -197,3 +200,26 @@ function onesignal_handle_quick_edit_date_change($post_id, $post, $update)
         onesignal_create_notification($post);
     }
 }
+
+// Function to cancel scheduled notifications when a post is deleted
+function onesignal_cancel_notification_on_post_delete($post_id)
+{
+    $post = get_post($post_id);
+
+    if (!$post) {
+        return;
+    }
+
+    if (!onesignal_is_post_type_allowed($post->post_type)) {
+        return;
+    }
```

Comment:
I recommend we remove this check as it is unnecessary and cases an edge-case.

The edge-casing being:
1. A post is scheduled that creates a schedule OneSignal push.
2. They change their OneSignal allowed post types.
3. The delete the post.
   * I would expect it to cancel the OneSignal push

Reply:
Good point, I hadn't considered that.